splunk_base/stackstate_checks/splunk/telemetry/splunk_telemetry_base.py

```diff
@@ -24,73 +40,42 @@ def __init__(self, name, init_config, agentConfig, persistence_check_name, insta
         self.status = None
         self.load_status()
 
+        # Hook to override instance creation
+    def _build_instance(self, instance):
+        return Instance(instance, self.init_config)
+
     def check(self, instance):
-        authentication = None
-        if 'url' not in instance:
-            raise CheckException('Splunk event/metric instance missing "url" value.')
-        if 'username' in instance and 'password' in instance and 'authentication' not in instance:
-            self.log.warning("This username and password configuration will be deprecated soon. Please use the new "
-                             "updated configuration from the conf")
-        if 'authentication' in instance:
-            authentication = instance["authentication"]
-            if 'basic_auth' not in authentication and 'token_auth' not in authentication:
-                raise CheckException('Splunk event/metric instance missing "authentication.basic_auth" or '
-                                     '"authentication.token_auth" value')
-            if 'basic_auth' in authentication:
-                basic_auth = authentication["basic_auth"]
-                if 'username' not in basic_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.basic_auth.username" '
-                                         'value')
-                if 'password' not in basic_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.basic_auth.password" '
-                                         'value')
-            if 'token_auth' in authentication:
-                token_auth = authentication["token_auth"]
-                if 'name' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.token_auth.name" value')
-                if 'initial_token' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing '
-                                         '"authentication.token_auth.initial_token" value')
-                if 'audience' not in token_auth:
-                    raise CheckException('Splunk event/metric instance missing "authentication.token_auth.audience" '
-                                         'value')
-
-        current_time = self._current_time_seconds()
-        url = instance["url"]
-        if url not in self.instance_data:
-            self.instance_data[url] = self.get_instance(instance, current_time)
-
-        instance = self.instance_data[url]
-        if not instance.initial_time_done(current_time):
-            self.log.debug("Skipping splunk metric/event instance %s, waiting for initial time to expire" % url)
-            return
+
+        if self.instance_data is None:
+            self.instance_data = self._build_instance(instance)
+
+        committable_state = CommittableState(self.commit_state, self.load_state(instance))
```

Comment:
commitable_state, this is how it is currently implemented in splunk_health and splunk_topology, do we need to change it because of transactional state